nixos/modules/programs/mepo.nix

```
@@ -0,0 +1,60 @@
+{ pkgs, config, lib, ...}:
+with lib;
+let
+  cfg = config.programs.mepo;
+in
+{
+  options.programs.mepo = {
+    enable = mkEnableOption (mdDoc "Fast, simple, hackable OSM map viewer");
+
+    locationBackends = {
+      gpsd = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc "Whether to enable location detection via gpsd.
+          This may require additional configuration of gpsd, see [here](#opt-services.gpsd.enable)";
+      };
+
+      geoclue = mkOption {
+        type = types.bool;
+        default = true;
+        description = mdDoc "Whether to enable location detection via geoclue";
+      };
+    };
+
+    menuSystem = mkOption {
+      type = types.enum [ "zenity" "dmenu" "bemenu" "framebuffer" ];
+      default = "zenity";
+      description = mdDoc "The menu program to use";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = with pkgs; [
+      mepo
+    ]
+    ++
+    lib.optional cfg.locationBackends.geoclue geoclue2-with-demo-agent
+    ++
+    lib.optional cfg.locationBackends.gpsd gpsd
+    ++
+    {
+      bemenu = [ bemenu ];
+      dmenu = [ dmenu xdotool ];
+      framebuffer = [ directfb libGL vis busybox ];
+      zenity = [ gnome.zenity ];
+    }.${cfg.menuSystem};
+
+    services.geoclue2 = mkIf cfg.locationBackends.geoclue {
+      enable = true;
+      appConfig.where-am-i = {
+        isAllowed = true;
+        isSystem = false;
+      };
+    };
+
+    services.gpsd.enable = mkIf cfg.locationBackends.gpsd true;
```

```suggestion
    services.gpsd.enable = cfg.locationBackends.gpsd;
```
we can simplify this